front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -0,0 +1,152 @@
+      start: mmToKm(position),
+    },
+    value:
+      identifier && sncf ? `${identifier.name} ${sncf.ch !== ('' || '00') ? sncf.ch : ''}` : '',
+  }));
+
+export const formatElectrifications = (
+  electrifications: PathPropertiesFormatted['electrifications']
+) =>
+  electrifications.map(({ electrificationUsage, start, stop }) => ({
+    position: {
+      start: mToKm(start),
+      end: mToKm(stop),
+    },
+    value: {
+      type: electrificationUsage.type,
+      voltage:
+        'voltage' in electrificationUsage
+          ? (electrificationUsage.voltage as ElectrificationValues['voltage'])
+          : undefined,
+      lowerPantograph:
+        'lower_pantograph' in electrificationUsage ? electrificationUsage.lower_pantograph : false,
+    },
+  }));
+
+export const formatSlopes = (slopes: PositionData<'gradient'>[]) => {
+  let actualSlopes: LayerData<number>[] = [];
+  if (slopes)
+    actualSlopes = slopes.map((slope, index) => ({
+      position: {
+        start: mToKm(index === 0 ? 0 : slopes[index - 1].position),
+        end: mToKm(slope.position),
+      },
+      value: slope.gradient,
+    }));
+  return actualSlopes;
+};
+
+export const getProfileValue = (
+  electricalProfileValue: SimulationResponseSuccess['electrical_profiles']['values'][number],
+  voltage?: '1500V' | '25000V'
+) => {
+  if (electricalProfileValue.electrical_profile_type === 'no_profile') {
+    return { electricalProfile: 'neutral' };
+  }
+  if (
+    !electricalProfileValue.handled ||
+    !electricalProfileValue.profile ||
+    electricalProfileValue.profile === null
+  ) {
+    return {
+      electricalProfile: 'incompatible',
+      color:
+        voltage === '1500V'
+          ? electricalProfilesDesignValues.F.color
+          : electricalProfilesDesignValues['20000V'].color,
+    };
+  }
+  return {
+    electricalProfile: electricalProfileValue.profile,
+    ...electricalProfilesDesignValues[
+      electricalProfileValue.profile as keyof typeof electricalProfilesDesignValues
+    ],
+  };
+};
+
+export const formatElectricalProfiles = (
+  simulation: SimulationResponseSuccess,
+  electrifications: Data['electrifications']
+) => {
+  if (simulation.electrical_profiles.values.length === 0) return undefined;
+  return simulation.electrical_profiles.values.map((values, index) => {
+    const electrification = electrifications.find(
+      (electrificationValue) =>
+        electrificationValue.position.start >=
+          (index === 0 ? 0 : mmToKm(simulation.electrical_profiles.boundaries[index - 1])) &&
+        electrificationValue.position.end! <=
+          mmToKm(simulation.electrical_profiles.boundaries[index])
+    );
+
+    return {
+      position: {
+        start: mmToKm(index === 0 ? 0 : simulation.electrical_profiles.boundaries[index - 1]),
+        end: mmToKm(simulation.electrical_profiles.boundaries[index]),
+      },
+      value: getProfileValue(values, electrification?.value.voltage as '1500V' | '25000V'),
```

Comment:
I don't think you need that ?

```suggestion
      value: getProfileValue(values, electrification?.value.voltage),
```

Reply:
oops, done